pkg/cmd/server/origin/master.go

```diff
@@ -582,6 +573,23 @@ func (c *MasterConfig) GetRestStorage() map[string]rest.Storage {
 		"clusterRoles":          clusterRoleStorage,
 	}
 
+	if sdnovs.IsOpenShiftNetworkPlugin(c.Options.NetworkConfig.NetworkPluginName) {
```

Comment:
Unfortunately these have always been enabled, we cannot disable them
without admin intervention.  With your code change, they would suddenly be
disabled when the admin upgraded.  That violates our backwards API
guarantees, where we don't change / break the behavior of the API without
admin intervention.

For 1.3 I don't think disabling these is worth the time or effort, since
it's unlikely users would use them.  If you want to make this better we
should document that users should delete stale resources when they change
plugins.

On Mon, Jun 13, 2016 at 3:13 PM, Ravi Sankar Penta <notifications@github.com
> wrote:

> In pkg/cmd/server/origin/master.go
> <https://github.com/openshift/origin/pull/9147#discussion_r66849632>:
>
> > @@ -582,6 +573,23 @@ func (c *MasterConfig) GetRestStorage() map[string]rest.Storage {
> >  		"clusterRoles":          clusterRoleStorage,
> >  	}
> >
> > +	if sdnovs.IsOpenShiftNetworkPlugin(c.Options.NetworkConfig.NetworkPluginName) {
>
> When user switches from openshift network plugin to third party plugin, oc
> get clusternetwork/hostsubnets provides stale info and it will be
> confusing(clusternetwork will show multitenant/subnet pluginName when they
> are using a different one). I didn't prefer to delete existing
> clusternetwork/hostsubnets because any unintentional typo in plugin name
> will wipe out the configuration.
> This is similar to storage for builds, build endpoints are enabled based
> on user config.These REST endpoints are specific to our SDN and if users
> are not using our plugin they shouldn't care about these endpoints?
> @smarterclayton <https://github.com/smarterclayton>
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/openshift/origin/pull/9147/files/816de37bfe1ff6cef3fbf81df8096bffa2d06030..06546b65a4c9375014c5d3ee8b80056afd73b78d#r66849632>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/ABG_p0s_pidYXnJUqVflXfqQ4RxmiwXrks5qLau8gaJpZM4Is66G>
> .
>